OutOfSchool/OutOfSchool.WebApi/Controllers/V1/AchievementController.cs

```diff
@@ -161,15 +182,61 @@ public async Task<ActionResult> Update(AchievementCreateDTO achievementDto)
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
     public async Task<IActionResult> Delete(Guid id)
     {
+        AchievementDto achievement;
+
+        try
+        {
+            achievement = await achievementService.GetById(id).ConfigureAwait(false);
+        }
+        catch (ArgumentException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+
+        var userHasRights = await this.IsUserProvidersOwnerOrAdmin(achievement.WorkshopId).ConfigureAwait(false);
+
+        if (!userHasRights)
+        {
+            return StatusCode(403, "Forbidden to delete achievement, which are not related to you");
+        }
+
         try
         {
             await achievementService.Delete(id).ConfigureAwait(false);
             return NoContent();
         }
-
         catch (ArgumentException ex)
         {
             return BadRequest(ex.Message);
         }
     }
+
+    private async Task<bool> IsUserProvidersOwnerOrAdmin(Guid workshopId)
```

Comment:
we have same method in some other controllers. Maybe create a base class so we don't ctrl+c/ctrl+v? :)

Reply:
that method is different